frontend/test/metabase/scenarios/onboarding/setup/setup.cy.spec.js

```diff
@@ -90,15 +90,39 @@ describe("scenarios > setup", () => {
         exact: false,
       });
 
+      // test database setup help card is NOT displayed before DB is selected
+      cy.findByTestId("database-setup-help-card").should("not.exist");
+
       // test that you can return to user settings if you want
       cy.findByText("Hi, Testy. Nice to meet you!").click();
       cy.findByLabelText("Email").should("have.value", "testy@metabase.com");
 
+      // test database setup help card is NOT displayed on other steps
+      cy.findByTestId("database-setup-help-card").should("not.exist");
+
       // now back to database setting
       cy.findByText("Next").click();
 
+      // check database setup card changes copy
+      cy.get("#formField-engine .AdminSelect").click();
+      popover()
+        .findByText("MySQL")
+        .click();
+      cy.findByTestId("database-setup-help-card").within(() => {
+        cy.findByText("Need help setting up MySQL?");
+        cy.findByRole("link", { name: /Our docs can help/i });
+      });
+
+      cy.get("#formField-engine .AdminSelect").click();
+      popover()
+        .findByText("SQLite")
+        .click();
+      cy.findByTestId("database-setup-help-card").findByText(
+        "Need help setting up your database?",
+      );
+
       // add h2 database
-      cy.findByText("Select a database").click();
+      cy.get("#formField-engine .AdminSelect").click();
```

Comment:
NIT: is it possible to use a test id here?

Reply:
Not sure. The form is generated by our entity framework and it also generates the input IDs. Not sure I can pass a test id to the input 🤔

Reply:
FWIW, it's perfectly fine to use `cy.findByText("Select a database")` since that's the **only** string like that on this page, while we can potentially have multiple `.AdminSelect` elements. Anyway, both work for this scenario so don't let this block you.

Reply:
Yeah, in the tests we change the engine a few times to ensure that the new help card changes texts and stuff. So I thought it would be better to use the same selector across the test, so it's more obvious we're clicking the same component a few times